packages/hatchify-core/src/dataTypes/number.ts

```diff
@@ -0,0 +1,132 @@
+import type {
+  ControlType,
+  HatchifyNumberProps,
+  PartialAttribute,
+} from "../types"
+
+export function number(props?: HatchifyNumberProps): PartialAttribute<number> {
+  const allowNull = props?.required != null ? !props.required : props?.required
+  const control: ControlType = {
+    type: "Number",
+    allowNull,
+    min: props?.min,
+    max: props?.max,
+    primary: props?.primary,
+    step: props?.step,
+  }
+
+  return {
+    name: `number(${props ? JSON.stringify(props) : ""})`,
+    orm: {
+      sequelize: {
+        type: "DOUBLE",
+        typeArgs: [],
+        allowNull,
+        autoIncrement: props?.autoIncrement,
+        primaryKey: props?.primary,
+        ...(props?.min || props?.max
+          ? {
+              validate: {
+                min: props?.min,
+                max: props?.max,
+              },
+            }
+          : {}),
+      },
+    },
+    control,
+    // Passed  - Any crazy value the client might send as a POST or PATCH
+    // Returns - A type the ORM can use
+    // Throws  - If the data is bad ❓
+    // Example : '2023-07-17T01:45:28.778Z' => new Date('2023-07-17T01:45:28.778Z')
+    //         : throw "'4 $core' is not a valid date";
+    setORMPropertyValue: (jsonValue: number | null): number | null => {
+      return clientMutationCoerce(jsonValue, control)
+    },
+
+    // Passed  - Any crazy STRING value the client might send as GET
+    // Returns - A type the ORM can use
+    // Throws  - If the data is bad ❓
+    // Example : ?filter[age]=xyz ... xyz => throw "xyz is not a number";
+    setORMQueryFilterValue: (queryValue: string): number | null => {
+      if (["null", "undefined"].includes(queryValue)) {
+        if (control.allowNull !== false) return null
+        throw new Error("Non-null value is required")
+      }
+
+      if (isNaN(queryValue as unknown as number)) {
```

Comment:
is this necessary to type it this way? I'm wondering if you can just coerce queryValue to be a number instead of using as unknown